controllers/src/main/scala/cromwell/pipeline/controller/ProjectFileController.scala

```diff
@@ -27,12 +27,30 @@ class ProjectFileController(wdlService: ProjectFileService)(implicit val executi
       },
       path("files") {
         post {
-          entity(as[ProjectUpdateFileRequest]) { request =>
-            onComplete(wdlService.uploadFile(request.project, request.projectFile, request.version)) {
-              case Success(Left(e)) => complete(StatusCodes.ImATeapot, e.getMessage) // TODO: change status code
-              case Success(_)       => complete(StatusCodes.OK)
-              case Failure(e)       => complete(StatusCodes.InternalServerError, e.getMessage)
-            }
+          entity(as[ProjectUpdateFileRequest]) {
+            request =>
+              val result = for {
+                validateResponse <- wdlService.validateFile(FileContent(request.projectFile.content))
+                uploadResponse <- wdlService.uploadFile(request.project, request.projectFile, request.version)
+              } yield {
+                if (validateResponse.isRight && uploadResponse.isRight) {
+                  StatusCodes.OK.intValue -> uploadResponse.right.get
+                } else if (validateResponse.isLeft && uploadResponse.isRight) {
+                  StatusCodes.Created.intValue -> uploadResponse.right.get
+                } else if (uploadResponse.isLeft) {
+                  StatusCodes.ImATeapot.intValue -> uploadResponse.left.get.message
+                }
+              }
+
+              while (!result.isCompleted) {}
```

Comment:
It's better to remove this string and replace ```result.value.get``` with ```onComplete(result)```

Reply:
onComplete function was implemented and line 45 was deleted